### PR TITLE
fixed:  No UI notifications on asking a query, and when receiving a reply

### PR DESCRIPTION
**Pull request by mehrab-alam**

### CODE REVIEW

File: src/components/LearnCourse/AboutCourse/CourseDiscussion/CourseDiscussion.tsx [hunk 1]
```diff
@@ -107,22 +108,34 @@ const QADiscssionTab: FC<{ resourceId: number; userId: string; loading: boolean
...
   };
+  const onPostQA = async (comment: string) => {
+    try {
+      DiscussionsService.postParentComment(
+        resourceId,
+        Number(router.query.courseId),
+        comment,
+        (result) => {
+          message.success("comment added");
+          fetchAllDiscussion();
```
Comment (amezng) [inline src/components/LearnCourse/AboutCourse/CourseDiscussion/CourseDiscussion.tsx:126]: Don't fetch all the discussions. Just append the newly added query/question

File: src/components/LearnCourse/AboutCourse/CourseDiscussion/CourseDiscussion.tsx [hunk 2]
```diff
@@ -107,22 +108,34 @@ const QADiscssionTab: FC<{ resourceId: number; userId: string; loading: boolean
...
+  const onClickMore = () => {
     if (totalCmt > pageSize) {
       const newPageSize = pageSize + 5;
       setPageSize(newPageSize);
       getAllDiscussioin(resourceId, newPageSize);
     }
   };
+  const onPostQA = async (comment: string) => {
+    try {
+      DiscussionsService.postParentComment(
```
Comment (amezng) [inline src/components/LearnCourse/AboutCourse/CourseDiscussion/CourseDiscussion.tsx:120]: Rename the method to `postQuery`

File: src/components/LearnCourse/AboutCourse/CourseDiscussion/CourseDiscussion.tsx [hunk 3]
```diff
@@ -107,22 +108,34 @@ const QADiscssionTab: FC<{ resourceId: number; userId: string; loading: boolean
...
 
   return (
     <section className={styles.qa_discussion_tab}>
-      <QAForm
-        loadingPage={loading}
-        resourceId={resourceId}
-        placeholder="Ask a Question"
-        fetchAllDiscussion={fetchAllDiscussion}
-      />
+      <QAForm loadingPage={loading} resourceId={resourceId} placeholder="Ask a Question" onPost={onPostQA} />
```
Comment (amezng) [inline src/components/LearnCourse/AboutCourse/CourseDiscussion/CourseDiscussion.tsx:138]: rename the method to `onQueryPost`

File: src/components/LearnCourse/AboutCourse/CourseDiscussion/ReplyDrawer.tsx [hunk 4]
```diff
@@ -71,6 +73,26 @@ const ReplyDrawer: FC<{
     }
   };
 
+  const onPostQA = async (comment: string) => {
```
Comment (amezng) [inline src/components/LearnCourse/AboutCourse/CourseDiscussion/ReplyDrawer.tsx:76]: Rename the method to `onPostReply`

File: src/components/LearnCourse/AboutCourse/CourseDiscussion/ReplyDrawer.tsx [hunk 5]
```diff
@@ -71,6 +73,26 @@ const ReplyDrawer: FC<{
...
+    try {
+      DiscussionsService.postReplyComment(
+        resourceId,
```
Comment (amezng) [inline src/components/LearnCourse/AboutCourse/CourseDiscussion/ReplyDrawer.tsx:79]: Use the method name to `postReply`

File: src/components/LearnCourse/AboutCourse/CourseDiscussion/ReplyDrawer.tsx [hunk 6]
```diff
@@ -71,6 +73,26 @@ const ReplyDrawer: FC<{
...
+        Number(router.query.courseId),
+        comment,
+        Number(sltComment?.id),
+        (result) => {
+          message.success("comment added ");
+          onCloseDrawer && onCloseDrawer();
+          fetchAllDiscussion();
```
Comment (amezng) [inline src/components/LearnCourse/AboutCourse/CourseDiscussion/ReplyDrawer.tsx:86]: Avoid calling the `fetchAllDiscussion`

File: src/pages/api/v1/discussions/add-reply-post.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
+import prisma from "@/lib/prisma";
+import { NextApiResponse, NextApiRequest } from "next";
+import { withMethods } from "@/lib/api-middlewares/with-method";
+import { withAuthentication } from "@/lib/api-middlewares/with-authentication";
+import { errorHandler } from "@/lib/api-middlewares/errorHandler";
+import { getToken } from "next-auth/jwt";
+import { getCookieName } from "@/lib/utils";
+
+/**
```
Comment (amezng) [inline src/pages/api/v1/discussions/add-reply-post.ts:9]: The API description should be `Post reply on a query`

File: src/pages/api/v1/discussions/post.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
+import prisma from "@/lib/prisma";
+import { NextApiResponse, NextApiRequest } from "next";
+import { withMethods } from "@/lib/api-middlewares/with-method";
+import { withAuthentication } from "@/lib/api-middlewares/with-authentication";
+import { errorHandler } from "@/lib/api-middlewares/errorHandler";
+import { getToken } from "next-auth/jwt";
+import { getCookieName } from "@/lib/utils";
+
+/**
+ * api for adding a comment
```
Comment (amezng) [inline src/pages/api/v1/discussions/post.ts:10]: change the description

File: src/services/DiscussionsService.ts [hunk 9]
```diff
@@ -25,8 +24,42 @@ type FailedApiResponse = {
...
+          result.json().then((r) => {
+            const apiResponse = r as ApiResponse;
+            onSuccess(apiResponse);
+          });
+        }
+      }
+    );
+  };
+
+  postReplyComment = (
```
Comment (amezng) [inline src/services/DiscussionsService.ts:51]: change method name to `postReply`

File: src/services/DiscussionsService.ts [hunk 10]
```diff
@@ -25,8 +24,42 @@ type FailedApiResponse = {
   error: string;
 };
 class DiscussionsSerivice {
-  addComment = (formData: any, onSuccess: (response: ApiResponse) => void, onFailure: (message: string) => void) => {
-    postWithFile(formData, `/api/v1/discussions/add/`).then((result) => {
+  postParentComment = (
```
Comment (amezng) [inline src/services/DiscussionsService.ts:27]: change method name to `postQuery`

**Review by amezng — CHANGES_REQUESTED** [review]

File: src/services/DiscussionsService.ts (hunk 9 above)
Comment (mehrab-alam) [reply src/services/DiscussionsService.ts:51]: resolved

File: src/pages/api/v1/discussions/add-reply-post.ts (hunk 7 above)
Comment (mehrab-alam) [reply src/pages/api/v1/discussions/add-reply-post.ts:9]: resolved

File: src/pages/api/v1/discussions/post.ts (hunk 8 above)
Comment (mehrab-alam) [reply src/pages/api/v1/discussions/post.ts:10]: resolved

File: src/components/LearnCourse/AboutCourse/CourseDiscussion/CourseDiscussion.tsx (hunk 3 above)
Comment (mehrab-alam) [reply src/components/LearnCourse/AboutCourse/CourseDiscussion/CourseDiscussion.tsx:138]: resolved

File: src/components/LearnCourse/AboutCourse/CourseDiscussion/ReplyDrawer.tsx (hunk 4 above)
Comment (mehrab-alam) [reply src/components/LearnCourse/AboutCourse/CourseDiscussion/ReplyDrawer.tsx:76]: resolved

File: src/components/LearnCourse/AboutCourse/CourseDiscussion/ReplyDrawer.tsx (hunk 5 above)
Comment (mehrab-alam) [reply src/components/LearnCourse/AboutCourse/CourseDiscussion/ReplyDrawer.tsx:79]: resolved

File: src/components/LearnCourse/AboutCourse/CourseDiscussion/CourseDiscussion.tsx (hunk 2 above)
Comment (mehrab-alam) [reply src/components/LearnCourse/AboutCourse/CourseDiscussion/CourseDiscussion.tsx:120]: resolved

**Review by amezng — APPROVED** [review]

**Pull request merged**